pkg/agent/openflow/client.go

```diff
@@ -916,3 +933,87 @@ func (c *client) IsIPv4Enabled() bool {
 func (c *client) IsIPv6Enabled() bool {
 	return config.IsIPv6Enabled(c.nodeConfig, c.encapMode)
 }
+
+// SendRejPacket generates reject packet as a packet-out and send it to OVS.
+func (c *client) SendRejectPacket(
```

Comment:
Is it better to define two funcs for ICMP and TCP respectively? The argument list is too long.

Reply:
And ICMPType/Code, etc. needs not be passed as they are just constants.

Or let SendRejectPacket() or two sub-funcs set all ICMP fields and call getTCPHeaderData().

Reply:
How about this: 
Add other func in client `func (c *client) GenBasePacketOutBuilder(srcMAC string, dstMAC string, srcIP string, dstIP string, inPort uint32, outPort uint32) PacketOutBuilder {}` and separate `SendRejectPacket()` to `SendTCPReject(pktOutBuilder PacketOutBuilder, TCPSrcPort uint16, TCPDstPort uint16, TCPSeqNum uint32, TCPAckNum uint32,)` and `SendICMPReject(pktOutBuilder PacketOutBuilder, ICMPData []byte, isIPv6 bool)`?  
The new func `GenBasePacketOutBuilder()` can also be used in `SendTraceflowPacket()` to shorten the args list. 
Or I can keep `TCPFlags` and `ICMPType/Code` args and change the func name to `SendTCPPacketOut()` and `SendICMPPacketOut()` to make it reuseable in the future.

Reply:
Not sure if I got what you mean. But why not change as what in you mind and we can see.

Reply:
Changed.